### PR TITLE
fix: make metric names unique, so that the same metric name isn't used

### PR DESCRIPTION
**Pull request by hstojic**

### CODE REVIEW

File: trieste/acquisition/optimizer.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def improvements() -> tf.Tensor:
                 if V == 1:
                     logging.scalar("spo_improvement_on_initial_samples", improvements)
                 else:
-                    logging.histogram("spo_improvement_on_initial_samples", improvements)
+                    logging.histogram("spo_improvement_on_initial_samples_across_subspaces", improvements)
```
Comment (hstojic) [inline trieste/acquisition/optimizer.py]:
```suggestion
                    logging.histogram("spo_improvements_on_initial_samples", improvements)
```

File: trieste/acquisition/rule.py [hunk 2]
```diff
@@ -330,7 +330,7 @@ def acquire(
                             )
                         else:  # vectorized acquisition function
                             logging.histogram(
-                                f"EGO.acquisition_function/maximum_found[{i+1}]", values
+                                f"EGO.acquisition_function/maximums_found[{i+1}]", values
```
Comment (hstojic) [inline trieste/acquisition/rule.py:333]: a bit confused here - I would have thought that `[{i+1}]` bit is part of the name as well, and here the naming would be different than what we have above?

File: trieste/bayesian_optimizer.py [hunk 3]
```diff
@@ -941,7 +941,7 @@ def write_summary_query_points(
     if tf.rank(query_points) == 2:
         for i in tf.range(tf.shape(query_points)[1]):
             if len(query_points) == 1:
-                logging.scalar(f"query_points/[{i}]", float(query_points[0, i]))
+                logging.scalar(f"query_point/[{i}]", float(query_points[0, i]))
```
Comment (hstojic) [inline trieste/bayesian_optimizer.py:944]: I think you have the same in the ask-tell optimizer as well

**Review by hstojic — COMMENTED** [review]
looks ok to me - @uri-granta what do you think?

File: trieste/acquisition/rule.py (hunk 2 above)
Comment (uri-granta) [reply trieste/acquisition/rule.py:333]: I'm confused by your confusion :-) Both the scalar and historgram were previously called `maximum_found[{i+1}]`.

File: trieste/bayesian_optimizer.py (hunk 3 above)
Comment (alansaul) [reply trieste/bayesian_optimizer.py:944]: I think it re-uses this code so it's already covered :+1:

**Review by uri-granta — APPROVED** [review]
LGTM!

**Pull request merged**